File: Selenium/Activities/Activity15.java
package examples;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import io.github.bonigarcia.wdm.WebDriverManager;

public class Activity15 {

    public static void main(String[] args) {

        WebDriverManager.firefoxdriver().setup();

        WebDriver driver = new FirefoxDriver();

        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://training-support.net/webelements/dynamic-attributes");

        System.out.println("Page title is: " + driver.getTitle());

        WebElement fullName = driver.findElement(
                By.xpath("//input[starts-with(@id,'full-name')]"));

        WebElement email = driver.findElement(
                By.xpath("//input[contains(@id,'-email')]"));

        WebElement eventDate = driver.findElement(
                By.xpath("//input[contains(@name,'event-date')]"));

        WebElement details = driver.findElement(
                By.xpath("//textarea[contains(@id,'additional')]"));

        fullName.sendKeys("Hema Sundar");
        email.sendKeys("hema@example.com");
        eventDate.sendKeys("2026-06-20");
        details.sendKeys("Automation testing using Selenium");

        driver.findElement(By.xpath("//button[text()='Submit']")).click();

        WebElement message = wait.until(
                ExpectedConditions.visibilityOfElementLocated(
                        By.id("action-confirmation")));

        System.out.println("Success message: " + message.getText());

        driver.quit();
    }
}

File: Selenium/Activities/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()

wait = WebDriverWait(driver, 10)

driver.get("https://training-support.net/webelements/dynamic-attributes")

print("Page title:", driver.title)

full_name = driver.find_element(By.XPATH, "//input[starts-with(@id,'full-name')]")
email = driver.find_element(By.XPATH, "//input[contains(@id,'-email')]")
event_date = driver.find_element(By.XPATH, "//input[contains(@name,'event-date')]")
details = driver.find_element(By.XPATH, "//textarea[contains(@id,'additional')]")

full_name.send_keys("Hema Sundar")
email.send_keys("hema@example.com")
event_date.send_keys("2026-06-20")
details.send_keys("Automation with Selenium")

driver.find_element(By.XPATH, "//button[text()='Submit']").click()

message = wait.until(
    EC.visibility_of_element_located((By.ID, "action-confirmation"))
)

print("Success message:", message.text)

driver.quit()